test/demo/fake_evaluation_test.exs

```diff
@@ -0,0 +1,127 @@
+defmodule Wanda.Executions.FakeEvaluationTest do
+  use ExUnit.Case
+
+  import Wanda.Factory
+
+  alias Wanda.Catalog.{
+    Check,
+    Fact,
+    Value
+  }
+
+  alias Wanda.Executions.{
+    CheckResult,
+    Execution,
+    FakeEvaluation,
+    Result
+  }
+
+  describe "complete_fake_execution/4" do
+    test "should complete a running fake execution" do
+      [
+        %Fact{
+          name: fact_name_1
+        },
+        %Fact{
+          name: fact_name_2
+        }
+      ] = catalog_facts = build_list(2, :catalog_fact)
+
+      [
+        %Value{
+          name: value_name_1
+        },
+        %Value{
+          name: value_name_2
+        }
+      ] = values = build_list(2, :catalog_value)
+
+      [
+        %Check{
+          id: check_id_1
+        },
+        %Check{
+          id: check_id_2
+        }
+      ] = checks = build_list(2, :check, facts: catalog_facts, values: values)
+
+      [
+        %Execution.Target{
+          agent_id: agent_id
```

Comment:
Can we add at least other agent?
At the end, the major part of the logic consists on doing things for 2 agents, which this doesn't cover